stix2/pattern_expressions.py

```diff
@@ -0,0 +1,223 @@
+from .constants import Constant, IntegerConstant, ListConstant, make_constant
+from .object_path import ObjectPath
+
+
+class PatternExpression(object):
```

Comment:
We should probably use the name `_Expression` here, or maybe just make the staticmethod a function?